cordova-lib/src/cordova/plugin.js

```diff
@@ -413,26 +385,55 @@ module.exports = function plugin(command, targets, opts) {
 
 function determinePluginTarget(projectRoot, cfg, target, fetchOptions) {
     var parsedSpec = pluginSpec.parse(target);
-
     var id = parsedSpec.package || target;
-
     // CB-10975 We need to resolve relative path to plugin dir from app's root before checking whether if it exists
     var maybeDir = cordova_util.fixRelativePath(id);
     if (parsedSpec.version || cordova_util.isUrl(id) || cordova_util.isDirectory(maybeDir)) {
         return Q(target);
     }
+    // Require project pkgJson.
+    var pkgJsonPath = path.join(projectRoot, 'package.json');
+    if(fs.existsSync(pkgJsonPath)) {
+        delete require.cache[require.resolve(pkgJsonPath)]; 
+        pkgJson = require(pkgJsonPath);
+    }
+
+    // If no parsedSpec.version, use the one from pkg.json or config.xml.
+    if (!parsedSpec.version) {
+        // Retrieve from pkg.json.
+        if(pkgJson && pkgJson.dependencies && pkgJson.dependencies[id]) {
+            parsedSpec.version = pkgJson.dependencies[id];
+            events.emit('verbose', 'No version specified for ' + id + ', retrieving version from package.json');
+        } else {
+            // If no version is specified, retrieve the version (or source) from config.xml.
+            parsedSpec.version = getVersionFromConfigFile(id, cfg);
+            events.emit('verbose', 'No version specified for ' + id + ', retrieving version from config.xml');
+        }
+    }
 
-    // If no version is specified, retrieve the version (or source) from config.xml
-    events.emit('verbose', 'No version specified for ' + parsedSpec.package + ', retrieving version from config.xml');
-    var ver = getVersionFromConfigFile(id, cfg);
+    // If parsedSpec.version satisfies pkgJson version, no writing to pkg.json. Only write when
+    // it does not satisfy.
+    if(parsedSpec.version) {
+        if(pkgJson && pkgJson.dependencies && pkgJson.dependencies[parsedSpec.package]) {
+            var noSymbolVersion;
+            if (parsedSpec.version.charAt(0) === '^' || parsedSpec.version.charAt(0) === '~') {
+                noSymbolVersion = parsedSpec.version.slice(1);
+            }
+            if (semver.satisfies(noSymbolVersion, pkgJson.dependencies[parsedSpec.package])) {
```

Comment:
if(!(semver.satisfies(noSymbolVersion, pkgJson.dependencies[parsedSpec.package])) and remove else